### ISSUE TITLE
Gender bias in model-generated content

### ISSUE THREAD

**Issue by Albertchamberlain** [Francis-Rings/StableAnimator#5]
The process should involve using character images, poses, and facial frames to animate the character in any selected image to perform the appropriate actions. However, when I executed it, I noticed that the character's gender changed.

For example:  
Using the pose and facial expression from case-1, and referencing the following image:

![martin](https://github.com/user-attachments/assets/eba4692d-f2cf-4dad-94df-04170864e824)

I achieved the following results:  
![animation_video](https://github.com/user-attachments/assets/481d1198-9491-4436-8d7b-f9cd62c9d32e)

Could you also tell me how the "lekun" in the readme was generated? : )

**Comment by Francis-Rings** [comment]
Hi! The likely issue is that the provided pose sequence is not aligned with the reference image. Skeleton misalignment can significantly impact the performance of animation models. To address this, you should apply a skeleton alignment algorithm to the pose sequence before using StableAnimator. Additionally, I recommend you to ensure the sizes of the given pose sequence and the reference image to be the same, which is relatively easier to apply the alignment algorithm. For details on alignment algorithms, you may refer to the human skeleton extraction section in the project’s README or consult resources like the [MimicMotion repository](https://github.com/Tencent/MimicMotion) or the [MotionEditor repository](https://github.com/Francis-Rings/MotionEditor).

**Comment by Albertchamberlain** [comment]
Can i use align.py file from MotionEditor project,like this ?

**Comment by Francis-Rings** [comment]
> Can i use align.py file from MotionEditor project,like this ?

Hi! I recommend first using the pose extraction code available in the StableAnimator repository. Detailed instructions can be found in the README file. By the way, I also attempted to animate the provided reference image, and the comparison results are as follows.

https://github.com/user-attachments/assets/a8c34091-4735-44a4-a0e2-3341c76d6cff

**Comment by Albertchamberlain** [comment]
WoW,Thx a lot

**Comment by Albertchamberlain** [comment]
> > Can i use align.py file from MotionEditor project,like this ?
> 
> Hi! I recommend first using the pose extraction code available in the StableAnimator repository. Detailed instructions can be found in the README file. By the way, I also attempted to animate the provided reference image, and the comparison results are as follows.嗨！我建议首先使用StableAnimator库中提供的姿势提取代码。详细说明可以在README文件中找到。顺便说一下，我也尝试对提供的参考图像进行动画处理，比较结果如下。关闭
> 
>  issue_comparison.mp4

Can you tell me how you got this result step by step?  ：）

**Comment by Francis-Rings** [comment]
> > > Can i use align.py file from MotionEditor project,like this ?
> > 
> > 
> > Hi! I recommend first using the pose extraction code available in the StableAnimator repository. Detailed instructions can be found in the README file. By the way, I also attempted to animate the provided reference image, and the comparison results are as follows.嗨！我建议首先使用StableAnimator库中提供的姿势提取代码。详细说明可以在README文件中找到。顺便说一下，我也尝试对提供的参考图像进行动画处理，比较结果如下。关闭
> > issue_comparison.mp4
> 
> Can you tell me how you got this result step by step? ：）

Hi, please refer to the Human Skeleton Extraction section in the README file for detailed instructions. The Human Skeleton Extraction Code I provided can automatically align the given pose sequence with the reference image. The inputs of the Human Skeleton Extraction only contains a given reference image and a driven MP4 video (or a folder containing multiple .png files).

### Human Skeleton Extraction
We leverage the pre-trained DWPose to extract the human skeletons. In the initialization of DWPose, the pretrained weights should be configured in `/DWPose/dwpose_utils/wholebody.py`:
```
onnx_det = 'path/checkpoints/DWPose/yolox_l.onnx'
onnx_pose = 'path/checkpoints/DWPose/dw-ll_ucoco_384.onnx'
```
Given the target image folder containing multiple .png files, you can use the following command to obtain the corresponding human skeleton images:
```
python skeleton_extraction.py --target_image_folder_path="path/test/target_images" --ref_image_path="path/test/reference.png" --poses_folder_path="path/test/poses"
```
It is worth noting that the .png files in the target image folder are named in the format `frame_i.png`, such as `frame_0.png`, `frame_1.png`, and so on. 
`--ref_image_path` refers to the path of the given reference image. The obtained human skeleton images are saved in `path/test/poses`. It is particularly significant that the target skeleton images should be aligned with the reference image regarding the body shape.

If you only have the target MP4 file (target.mp4), we recommend you to use `ffmpeg` to convert the MP4 file to multiple frames (.png files) without any quality loss.
```
ffmpeg -i target.mp4 -q:v 1 path/test/target_images/frame_%d.png
```
The obtained frames are saved in `path/test/target_images`.

**Comment by Yaqing2023** [comment]
but for the existing poses like case-1, there is no target_images, so if i want to use a new image as reference image， 后卫头润skeleton_extraction.py for alignment?

**Comment by Francis-Rings** [comment]
> @Francis-Rings but for the existing poses like case-1, there is no target_images, so if i want to use a new image as reference image， how to run skeleton_extraction.py for alignment?

Hi, you may use the existing animated videos as driven videos. You can just conduct pose alignment between your new reference image and the existing animated video.

**Comment by Yaqing2023** [comment]
@Francis-Rings I see. So the routine is 1: use the existing reference image and poses to generate a video 2：extract frames from the video to target_images directory 3: use this to do alignment for new reference image 4: no need for face extract again
thanks

**Comment by Francis-Rings** [comment]
> @Francis-Rings I see. So the routine is 1: use the existing reference image and poses to generate a video 2：extract frames from the video to target_images directory 3: use this to do alignment for new reference image 4: no need for face extract again thanks

Yep, that’s correct!
